packages/accounts-controller/src/AccountsController.ts

```diff
@@ -241,12 +264,53 @@ export class AccountsController extends BaseController<
   }
 
   /**
-   * Returns the selected internal account.
+   * Returns the selected evm internal account by default unless the namespace is not eip155
    *
+   * @param chainId - Caip2 Id of the account
    * @returns The selected internal account.
    */
-  getSelectedAccount(): InternalAccount {
-    return this.getAccountExpect(this.state.internalAccounts.selectedAccount);
+  getSelectedAccount(chainId: Caip2ChainId = 'eip155:*'): InternalAccount {
+    if (!isCaipChainId(chainId) && chainId !== 'eip155:*') {
+      throw new Error(`Invalid CAIP2 id ${String(chainId)}`);
+    }
+
+    // TODO: have CAIP2 addresses within InternalAccount
+    const selectedAccount = this.getAccountExpect(
+      this.state.internalAccounts.selectedAccount,
+    );
+
+    if (
+      !chainId.startsWith('eip155:') ||
+      (chainId.startsWith('eip155:') &&
+        selectedAccount.type.startsWith('eip155:'))
```

Comment:
I wonder if we should have a `isAccountCompatibleWithChain` (or maybe a better name), so that we could use it in `listAccounts` and this function too, maybe something like:

```ts
function isAccountCompatibleWithChain(account: InternalAccount, chainNamespace: CaipNamespace): boolean {
  // TODO: Change this logic to not use account's type
  return account.type.startsWith(namespace);
}
```